controller/common/lut.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy import interpolate
+
+
+class LUT:
+    """
+    Class for performing look-up table interpolation.
+
+    Methods:
+        __init__: Initializes the LUT object with lookup table data and interpolation method.
+        __call__: Calls the interpolation method with the given input.
+    """
+
+    def __init__(self, lookup_table: list, interpolation_method: str = "linear"):
+        """
+        Initializes the LUT object.
+
+        Args:
+            lookup_table (list): A list of tuples or lists containing x-y data points for
+            interpolation. Format is [[x-values],[y-values]]
```

Comment:
I think the format should be `[[x1, y1], [x2, y2], ...]` based on your implementation

Reply:
updated formate:

```
        """
        Initializes the LUT object.

        Args:
            lookup_table (list): A list of tuples or lists containing x-y data points for
            interpolation. Format is [[x1,y1], [x2,y2], ..., [xn,yn]]
            interpolation_method (str): Interpolation method to use. Default is "linear".

```